src/pyodide/python-entrypoint-helper.js

```diff
@@ -173,27 +173,38 @@ async function preparePython() {
   return await getMainModule();
 }
 
-
-export default {
-  async fetch(request, env, ctx) {
+function makeHandler(pyHandlerName) {
+  return async function (...args) {
     try {
       const mainModule = await preparePython();
-      if (mainModule.on_fetch === undefined) {
-        throw new Error("Python Worker should define an on_fetch method");
-      }
-      return await mainModule.on_fetch.callRelaxed(request, env, ctx);
+      return await mainModule[pyHandlerName].callRelaxed(...args);
     } catch (e) {
       console.warn(e.stack);
       throw e;
     }
-  },
-  async test(ctrl, env, ctx) {
-    try {
-      const mainModule = await preparePython();
-      return await mainModule.test.callRelaxed(ctrl, env, ctx);
-    } catch (e) {
-      console.warn(e.stack);
-      throw e;
+  };
+}
+
+const handlers = {};
+
+if (IS_WORKERD) {
+  handlers.fetch = makeHandler("on_fetch");
+  handlers.test = makeHandler("test");
+} else {
```

Comment:
why not do the same for workerd as below?

Reply:
It's the same reason we have the `IS_WORKERD` check above:
```
  // If we're in workerd, we have to do setupPackages in the IoContext, so don't start it yet.
  // TODO: fix this.
```
can't call `setupPackages` in workerd without an IoContext, can't import the main module without calling `setupPackages` first.

Reply:
not a blocker for this PR but we'll need to fix this to support other handlers in local dev mode

Reply:
Indeed.

Reply:
How long before we have on_scheduled in local mode along with triggers supported for c3 and deployment with reqs.txt supported for python workers ? @dom96  @hoodmane 

I need a tentative timeline for this. thanks in advance.